src/response.rs

```diff
@@ -155,14 +233,14 @@ impl Response {
     /// let string: String = res.body_string().await?;
     /// # Ok(()) }
     /// ```
-    pub async fn body_string(&mut self) -> Result<String, Error> {
+    pub async fn body_string(&mut self) -> http_types::Result<String> {
         let bytes = self.body_bytes().await?;
-        let mime = self.mime();
+        let mime = self.content_type();
         let claimed_encoding = mime
             .as_ref()
-            .and_then(|mime| mime.get_param("charset"))
-            .map(|name| name.as_str());
-        decode_body(bytes, claimed_encoding)
+            .and_then(|mime| mime.param("charset"))
+            .map(|name| name.to_string());
+        decode_body(bytes, claimed_encoding.as_deref())
```

Comment:
Neat; excited you managed to translate this. We should probably consider adding this to `http-types` eventually (believe it's the longest-open issue right now).

Reply:
It's a tad tricky there due to the absence of headers information from `Body`. maybe on `http_types::request` though.